Source/WebKit/UIProcess/Extensions/WebExtensionController.cpp

```diff
@@ -91,8 +91,8 @@ WebExtensionControllerParameters WebExtensionController::parameters() const
 WebExtensionController::WebProcessProxySet WebExtensionController::allProcesses() const
 {
     WebProcessProxySet processes;
-    for (auto& page : m_pages)
-        processes.add(page.legacyMainFrameProcess());
+    for (Ref page : m_pages)
+        processes.add(page->legacyMainFrameProcess());
```

Comment:
`legacyMainFrameProcess()` seems like a trivial getter, why do we need the `Ref`?

Reply:
I think it's the call to add() that's the problem. This is a limitation of scope-based rules. The analysis only sees that 'page' is in scope when we call add(); it doesn't do data flow to check whether 'page' is dead by the time we call add().